test/internet/test-dns.js

```diff
@@ -471,6 +471,92 @@ TEST(function test_lookup_localhost_ipv4(done) {
 });
 
 
+TEST(function test_lookup_ip_all(done) {
+  var req = dns.lookup('127.0.0.1', {all: true}, function(err, ips, family) {
+    if (err) throw err;
+    assert.ok(ips.length > 0);
+    assert.ok(Array.isArray(ips));
+    assert.strictEqual(ips[0].address, '127.0.0.1');
+    assert.strictEqual(ips[0].family, 4);
+
+    done();
+  });
+
+  checkWrap(req);
+});
+
+
+TEST(function test_lookup_null_all(done) {
+  var req = dns.lookup(null, {all: true}, function(err, ips, family) {
+    if (err) throw err;
+    assert.strictEqual(ips.length, 0);
+    assert.ok(Array.isArray(ips));
```

Comment:
It doesn't matter a ton since this is a test, but I think you'd want to check for `Array.isArray(ips)` before checking the `ips.length`.